lambda-buffers-codegen/src/LambdaBuffers/Codegen/Typescript/Print/Ty.hs

```diff
@@ -89,7 +89,10 @@ printSum pkgMap tyN (PC.Sum ctors _) = do
   return $
     group $
       if null ctors
-        then mempty
+        then -- TODO(jaredponn): is this what we want? The empty sum
+        -- corresponds to the `never` type in the sense that there are no
+        -- values which satisfy this type
+          "never"
```

Comment:
Before, this would generate invalid code e.g.
```ts
export type MySum =
```
is syntactically invalid TS, so I changed this to `never`. This is probably what we mean by the "empty sum"

Reply:
Yoop! I think I need to revisit the empty sum/record/product and have a coherent story throughout the pipeline. I don't really know how i treat it tbh

Reply:
I actually wanted to have a compiler pass that calculates cardinality of LambdaBuffers types, I think that can be a very useful feature. And any 0 cardinality, due to empty sum for example, or recursion `prod X = X` would cause a compiler error.

It's a nice problem to model and solve, so that would mean that we'd never reach codegen with empty LB types? Does that make sense?

Reply:
I see! That does make sense! I'll leave it as the `never` type in the meantime!